File: c/src/common/honoluluclamchowder.c

#include "honoluluclamchowder.h"

soup_t * honoluluclamchowder_new()
{
	soup_t * soup = soup_new();
	soup->name = "PacificClamChowder";
	arraylist_string_add( soup->ingredients,"1 Pound Fresh Pacific Clams");
	arraylist_string_add( soup->ingredients,"1 cup pineapple chunks");
	arraylist_string_add( soup->ingredients,"1/2 cup coconut milk");
	arraylist_string_add( soup->ingredients,"1/4 cup SPAM");
	arraylist_string_add( soup->ingredients,"1/4 cup taro chips");
   return soup;
}

File: clojure/src/common/honoluluclamchowder.clj
(defclass honoluluclamchowder
   (clamchowder)
   ())

(defmethod initialize-instance :after ((this honoluluclamchowder) &key)
   (setf (slot-value this 'name) "PacificClamChowder")
   (setf (slot-value this 'ingredients) (list
        "1 Pound Fresh Pacific Clams"
        "1 cup pineapple chunks"
        "1/2 cup coconut milk"
        "1/4 cup SPAM"
        "1/4 cup taro chips"
	)))

